Various/hemisphera_Zuljeah/Zuljeah_Monitor.lua

```diff
@@ -0,0 +1,153 @@
+-- @noindex
+
+local function collectRegions()
+  local regions = {}
+  markerCount = reaper.CountProjectMarkers(0)
+  for i = 0,markerCount-1 do
+    ok, isRegion, posStart, posEnd, name, id = reaper.EnumProjectMarkers2(0, i)
+    if (isRegion) then
+      local r = {}
+      r["id"] = id
+      r["name"] = name
+      r["sequence"] = 0
+      r["start"] = posStart
+      r["end"] = posEnd
+      regions[i] = r
+    end
+  end
+  return regions
+end
+
+local function findSetlistTrack()
+  local trackCount = reaper.GetNumTracks()
+  for i = 0, trackCount - 1 do
+    local track = reaper.GetTrack(0, i)
+    local isMuted = reaper.GetMediaTrackInfo_Value(track, 'B_MUTE')
+    local ok, trackName = reaper.GetTrackName(track)
+    if (isMuted == 0) and (string.find(trackName, "SETLIST")) then
+      return track
+    end
+  end
+  return nil
+end
+
+local function buildSetlist(track)
+  if (track == nil) then
+    track = findSetlistTrack()
+  end
+  if (track == nil) then
+    return nil
+  end
+
+  local itemCount = reaper.GetTrackNumMediaItems(track)
+  local regions = collectRegions()
+  sq = {}
+  local sortedRegions = {}
+  for i = 0, itemCount - 1 do
+    local mi = reaper.GetTrackMediaItem(track, i)
+    local ok, take = reaper.GetSetMediaItemInfo_String(mi, "P_NOTES", "", false)
+    local pos = reaper.GetMediaItemInfo_Value(mi, 'D_POSITION')
+    local seq = tonumber(tostring(take))
+    for k,v in pairs(regions) do
+      if (v["start"] <= pos) and (v["end"] >= pos) then
+        v["mediaitem"] = mi
+        v["selected"] = reaper.IsMediaItemSelected(mi)
+        sortedRegions[seq] = v
```

Comment:
Sorry for the long delay... Tried it out: it crashes on this line (seq is nil) if `P_NOTES` fails to parse as a number.

The 'Consolas' font is only pre-installed on Windows.

The user-facing actions should have the author prefix and be named as sentence case (eg. "hemisphera_Zuljeah - Next song.lua") to blend well with other actions.

Since all three Lua files share a lot of code, I recommend putting the common code in one file and loading that from the others instead of duplicating. A good place would be the hemisphera_Zuljeah.lua package file (in which case it needs to be marked for installation, but not listed in the action list).

Is this behavior normal when starting playback after moving the cursor a bit before the regions?
![Take](https://github.com/user-attachments/assets/03154c63-1cf1-43c2-a903-0a52de27da06)

Docker #0 may be anywhere (top/bottom/left/right/floating), so it would be best to allow the user to choose in which docker to put the window.